doc/s3ip_sysfs/s3ip_sysfs_hld.md

```diff
@@ -0,0 +1,309 @@
+ - The sysfs path defined in this specification must exist, and the file content should be "NA" if no hardware is available.
+
+S3IP sysfs specifiction : [specifiction](/doc/s3ip_sysfs/s3ip_sysfs_specification.md)
+
+S3IP sysfs specification hierarchy overview
+- sys_switch
+  - cpld
+    - number
+    - cpld1
+       - alias
+       - type
+       - ...
+  - psu
+  - syseeprom
+  - sysled
+  - transceiver
+  - temp_sensor
+  - vol_sensor
+  - cur_sensor
+  - watchdog
+  - slot
+  - fpga
+
+
+### 3. S3IP sysfs framework
+#### 3.1 S3IP sysfs framework scenario
+The goal of the S3IP sysfs framework is to make it easy for vendors to generate sysfs directory structures that conform to the S3IP sysfs specification. 
+
+![framework_scenario](/images/s3ip_sysfs/s3ip_sysfs_framework_scenario.svg)
+
+S3IP sysfs framework contains two parts。
+ - S3ip_sysfs_service: kernel module management, configuration file parsing and /sys_switch directory generation.
+ - S3ip_sysfs_framework: the framework kernel module.
+
+#### 3.2 S3IP sysfs framework design
+Design Principles
+ - Keep kernel modules as simple as possible.
+ - Reuse sysfs paths of existing drivers as much as possible.
+ - Easy to implement and debug.
+
+The process for creating /sys_switch/
+
+![setup_flow](/images/s3ip_sysfs/s3ip_sysfs_setup_flow.svg)
+
+Flow to get the fan speed
+
+![get_flow](/images/s3ip_sysfs/s3ip_sysfs_get_info_flow.svg)
+
+ the code for submission:
+ - Two device platforms that conform to the S3IP sysfs specification will be submitted to the buildimage repository
+ - S3ip sysfs framework will be submitted to the sonic-platform-common repository 
+
+##### 3.2.1 s3ip sysfs kernel module
+
+The specification is not only a guideline, but also contains requirements for implementation. The kernel module is designed to make it easy to meet those details. 
+
+S3IP sysfs kernel module(aka s3ip_sysfs.ko) provided the following features:
+
+1. Register/unregister mechanisms to interact with the s3ip_sysfs.ko.
+
+2. Dynamically generate/destroy the corresponding directory when a driver that uses s3ip_sysfs.ko is installed/uninstalled.  
+
+Sample code for the watchdog interactive interface, The s3ip_sysfs.ko implements this code
+
+```
+struct s3ip_sysfs_watchdog_drivers_s {
+    ssize_t (*get_watchdog_identify)(char *buf, size_t count);
+    ssize_t (*get_watchdog_state)(char *buf, size_t count);
+    ssize_t (*get_watchdog_timeleft)(char *buf, size_t count);
+    ssize_t (*get_watchdog_timeout)(char *buf, size_t count);
+    int (*set_watchdog_timeout)(int value);
+    ssize_t (*get_watchdog_enable_status)(char *buf, size_t count);
+    int (*set_watchdog_enable_status)(int value);
+};
+
+extern int s3ip_sysfs_watchdog_drivers_register(struct s3ip_sysfs_watchdog_drivers_s *drv);
+extern void s3ip_sysfs_watchdog_drivers_unregister(void);
+ 
+```
+
+Sample code exposes watchdog register to sysfs, This code should be implemented by the vendor

```

Comment:
Vendors can have drivers which are common to various NOS. Isn't the S3IP framework mandating them to create SONiC specific drivers?

Reply:
We consider the case of the common driver, so by the configuration file, the path of the common driver, soft link to S3IP sysfs directory, do not need to create SONiC specific drivers